Libs/Visualization/VTK/Widgets/Testing/Cpp/ctkVTKAbstractViewTest1.cpp

```diff
@@ -0,0 +1,110 @@
+/*=========================================================================
+
+  Library:   CTK
+
+  Copyright (c) Kitware Inc.
+
+  Licensed under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+
+      http://www.apache.org/licenses/LICENSE-2.0.txt
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+
+=========================================================================*/
+
+// Qt includes
+#include <QApplication>
+#include <QDebug>
+#include <QTimer>
+
+// CTK includes
+#include "ctkCommandLineParser.h"
+#include "ctkVTKSliceView.h"
+#include "ctkVTKRenderView.h"
+
+// VTK includes
+#if CTK_USE_QVTKOPENGLWIDGET
+#include <QVTKOpenGLWidget.h>
+#endif
+#include <vtkRenderWindow.h>
+#include <vtkCallbackCommand.h>
+
+unsigned int RenderCount = 0;
+
+//-----------------------------------------------------------------------------
+void onRenderEvent(vtkObject *caller, unsigned long vtkNotUsed(eid), void *clientData, void *vtkNotUsed(callData))
+{
+  ++RenderCount;
+}
+
+//-----------------------------------------------------------------------------
+int ctkVTKAbstractViewTest1(int argc, char * argv [] )
+{
+#if CTK_USE_QVTKOPENGLWIDGET
+    QSurfaceFormat format = QVTKOpenGLWidget::defaultFormat();
+    format.setSamples(0);
+    QSurfaceFormat::setDefaultFormat(format);
+#endif
+
+  QApplication app(argc, argv);
+
+  // Command line parser
+  ctkCommandLineParser parser;
+  parser.addArgument("", "-I", QVariant::Bool);
+  QHash<QString, QVariant> parsedArgs = parser.parseArguments(app.arguments());
+  bool interactive = parsedArgs["-I"].toBool();
+
+  // Instantiate slice view
+  ctkVTKSliceView sliceView;
+  sliceView.setHighlightedBoxColor(QColor(Qt::yellow));
+  sliceView.setCornerAnnotationText("SliceView");
+  
+  vtkNew<vtkCallbackCommand> renderEventCallback;
+  renderEventCallback->SetCallback(onRenderEvent);
+  sliceView.renderWindow()->AddObserver(vtkCommand::RenderEvent, renderEventCallback);
+  sliceView.setMaximumUpdateRate(INFINITE);
+  sliceView.show();
+
+  sliceView.scheduleRender();
+  Sleep(100);
+  sliceView.scheduleRender();
+
+  if (RenderCount == 0)
+  {
+    qDebug() << "Missing render event!";
+    return EXIT_FAILURE;
+  }
+
+  bool sliceViewWasPaused = sliceView.pauseRender(true);
+  RenderCount = 0;
+
+  sliceView.scheduleRender();
+  Sleep(100);
+  sliceView.scheduleRender();
+  
+  if (RenderCount != 0)
+  {
+    qDebug() << "Invalid render event!";
+    return EXIT_FAILURE;
+  }
+
+  sliceView.pauseRender(sliceViewWasPaused);
```

Comment:
Add test where a function calls pauseRender(true), then calls another function that uses pauseRender. This would be good both for testing the behavior and as a usage example.